### DISCUSSION TITLE
Add support for custom caret and padding

### DISCUSSION THREAD

**Discussion by ghost** [PhilippeMarcMeyer/vanillaSelectBox#25] [Ideas]
Hi. It'd be nice if you could add support for a custom caret icon and padding.

**Comment by PhilippeMarcMeyer** [comment]
I'm not sure about what to do : 

![image](https://user-images.githubusercontent.com/2178299/104085937-8dd14f00-5253-11eb-8810-aaccb746f25b.png)

Before, this top bottom , had just a class to define its style.
The problem is that the css was  then often overrun by other css defined in the web page
so I decided for an in-line css with de proprietary style attribute.

The solution is maybe to allow you to send some css as object is the params ?

Would it be OK ?

Here are the proprietary css values : 

```
       cssArray= [
                {"key":"min-width","value":"120px"},
                {"key":"border-radius","value":"0"},
                {"key":"width","value":"100%"},
                {"key":"text-align","value":"left"},
                {"key":"z-index","value":"1"},
                {"key":"color","value":"#333"},
                {"key":"background","value":"white !important"},
                {"key":"border","value":"1px solid #999 !important"},
                {"key":"line-height","value":"20px"},
                {"key":"font-size","value":"14px"},
                {"key":"padding","value":"6px 12px"}
                ]
        }

```

About the caret : 

`<span class="caret" style="position: absolute; right: 8px; margin-top: 8px;"></span>`

You can see it is a span tag using some inline style and a "caret" class.

This class is defined in vanillaSelectBox.css

```
.caret {
	display: inline-block;
	width: 0;
	height: 0;
	margin-left: 2px;
	vertical-align: middle;
	border-top: 4px dashed;
	border-top: 4px solid;
	border-right: 4px solid transparent;
	border-left: 4px solid transparent;
}
```

you may change it

Let me know what you think

Thank you for opening this discussion

I apreciate that

Philippe Marc

**Comment by ghost** [comment]
Ah hello. Thanks for the response. I'll change the caret class. Though concerning the checkboxes for the multiple select, how would you change those?

**Comment by PhilippeMarcMeyer** [comment]
Play with :
.vsb-menu .multi li.active::after (the check sign)
.vsb-menu .multi li.active::before (the check box)

**Reply by PhilippeMarcMeyer** [reply]
Please try the special branch : "freeStyledButton" 

Change .vsb-main button in the css file to your liking
```
.vsb-main button{
	min-width: 120px;
	border-radius: 0;
	width: 100%;
	text-align: left;
	z-index: 1;
	color: #333;
	background: white !important;
	border: 1px solid #999 !important;
	line-height:20px;
	font-size:14px;
	padding:6px 12px;
}

```
the program should take your values and add then to a style attribute + default ones if they are missing in the css file.

The goal is to offer the possibility to change the style but keep a css protection against inherited styles from the guest page.

**Reply by ghost** [reply]
Ah alright. Thanks very much for your help!
